zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ProcessInstanceController.java

```diff
@@ -20,8 +29,31 @@ public class ProcessInstanceController {
 
   @PostMapping(
       path = "/process-instances/search",
-      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE})
-  public ResponseEntity<Object> searchProcessInstances() {
-    return ResponseEntity.ok(processInstanceServices.search((q) -> q));
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<ProcessInstanceSearchQueryResponse> searchProcessInstances(
+      @RequestBody final ProcessInstanceSearchQueryRequest query) {
+    return SearchQueryRequestMapper.toProcessInstanceQuery(query)
+        .fold(this::search, RestErrorMapper::mapProblemToResponse);
+  }
+
+  private ResponseEntity<ProcessInstanceSearchQueryResponse> search(
+      final ProcessInstanceQuery query) {
+    try {
+      final var tenantIds = TenantAttributeHolder.tenantIds();
+      final var result =
+          processInstanceServices.withAuthentication((a) -> a.tenants(tenantIds)).search(query);
+      return ResponseEntity.ok(
+          SearchQueryResponseMapper.toProcessInstanceSearchQueryResponse(result).get());
```

Comment:
❓ Just out of curiosity: would async searching be possible? Currently, this is a synchronous blocking call in the service's API. Have we discussed this anywhere in the past?

**Note**: this is not blocking the merge at all.

Reply:
At least, I am not aware of whether this has been discussed or if any decision is being made. Technically, yes, we could also implement them asynchronously, ES and OS supports this as well.

Reply:
OK, let's defer that. Could be an interesting adjustment though to avoid blocking search requests and free I/O resources for search requests over REST.